src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,109 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
```

Comment:
Hmm, when we were discussing this, I thought we were talking specifically about the concept of an `Income`, which in our version of the world is always represented with a current and future value. I'm not sure if this concept is as universal when applied to something as a generic as a form value.

If this refers to the difference between the input's current value vs. some previous value, then that seems like a different concept, better handled by our view layer (React) than by our data models.

If we're storing their last, fully valid input before they moved on to another input, maybe that's a case for this. I'm not positive what our objective is here.

Reply:
There are a lot of future/past/present concepts in there, so I'll try to explain the situation. *All* our form values have a theoretical current and projected/imagined future value, just like income. This, again, depends partially on how we want to think of this project moving forward. Theoretically, in future you'd be able to say, 'What if I got pregnant? What would be my benefit situation in that case?'. There would eventually be two stages to the form - one for all current values, and one for all projected/future values. Does that clarify a bit?

Reply:
That helps a lot. Thank you.

I am curious about the *all* though - we have non-temporal form data: when a client enters their name, or when they select that they use SNAP benefits. These are temporally static form values. Is there a more scoped name we could give to `FormValue` to indicate that it only refers to the subset of data we track that has a past and future state?

Reply:
Good point, though their name isn't one of the questions in the form. Any ideas for a better name?